remove interactive prompts and add pgpassword

hcavalle opened the pull request:
Removes all interactive prompts and adds pg password to `security.yml` process.

ChrisMcKenzie commented:
@hcavalle I am really excited about this I have been wanting to dig in to this script for a while 👍

ChrisMcKenzie commented on security-setup:
```diff
@@ -689,34 +708,38 @@ class DockerRegistry(Component):
+                                prompt='Docker registry password: ')
+                        email = os.environ.get('DOCKER_EMAIL')
+                        if not email:
+                            email = 'admin@datascience.com'
```
This move makes our mantl fork highly specialized. I am ok with this just want to point out if we ever decide to PR to upstream

hcavalle replied:
yeah I got lazy at that point! I moved it out to an additional flag on the parser, but still defaults to `admin@datascience` but now it's 'easy' to pinpoint and swap out.

tphummel commented:
@hcavalle sounds good. what would a non-interactive usage after these changes look like?

hcavalle commented:
`make create-secrets`:
```
../../.roles/mantl-roles/security-setup --use-private-docker-registry True
============= Certificates =============
----> certificate authority
root CA already exists
================ Nginx =================
----> admin password
set nginx admin password
============== Postgresql ==============
----> dsadmin password
set postgresql admin password
================ Consul ================
----> check security
----> gossip key
set gossip key
----> master acl token
set acl master token
----> agent acl token
set acl agent token
----> secure acl token
set acl secure token
=============== Marathon ===============
----> check security
----> marathon framework authentication
set marathon framework secret
----> admin password
set marathon http credentials
----> marathon acl token
set acl marathon token
================ Vault =================
----> vault acl token
set acl vault token
=============== Chronos ================
----> check security
----> chronos framework authentication
set chronos framework secret
----> admin password
set chronos http credentials
=============== MantlApi ===============
----> check security
----> Mesos authentication for frameworks installed via Mantl API
set Mantl API framework secret
----> mantl-api acl token
set acl mantl-api token
============ DockerRegistry ============
----> None
set docker registry credentials
============== Zookeeper ===============
----> super user auth
set zk super user secret
----> mesos user auth
set zk mesos user secret
----> marathon user auth
set zk marathon user secret
----> chronos user auth
set zk chronos user secret
=============== MantlUI ================
----> check security
================ Mesos =================
----> check security
----> framework auth
set auth for Marathon
set auth for Chronos
set auth for Mantl API
----> follower auth
enabled follower auth
----> remove deprecated settings
========================================
Wrote security settings to /home/hcavalle/ds/repos/ds/multi-cloud/.roles/mantl-roles/security.yml. Include them in your Ansible run like this:

    ansible-playbook your-playbook.yml -e @/home/hcavalle/ds/repos/ds/multi-cloud/.roles/mantl-roles/security.yml
# move security yaml so that subsequent secrets are not duplicated.
mv ../../.roles/mantl-roles/security.yml .
cp -r ../../.roles/mantl-roles/ssl .
```

hcavalle commented:
we'd want to change the template in multi-cloud to not have the `postgres_password` value any longer as well I think. to retrofit existing customer envs, we'd have to do some cleanup between the two.

hcavalle commented:
I also just realized if you run `make create-secrets` more than once it overwrites the existing password. should add a check that the PW exists or not since that could cause problems, right?

hcavalle commented:
updated with a change to the email being hardcoded in the component itself. and fixed the postgres_password component to not overwrite on re-run if it has already been set.
